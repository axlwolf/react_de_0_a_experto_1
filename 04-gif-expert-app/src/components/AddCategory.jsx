/* eslint-disable react/prop-types */
import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState("");

	const handleInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();

		if (inputValue.trim().length <= 2) return;

		onNewCategory(inputValue.trim());
		setInputValue("");
	};

	return (
		<form onSubmit={handleSubmit}>
			<input
				type="text"
				placeholder="Find gifs"
				value={inputValue}
				onChange={(e) => handleInputChange(e)}
			/>
		</form>
	);
};