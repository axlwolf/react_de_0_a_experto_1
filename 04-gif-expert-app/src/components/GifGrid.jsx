/* eslint-disable react/prop-types */
import { GifItem } from "./GifItem";

import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
	const { gifs, isLoading } = useFetchGifs(category);

	return (
		<>
			<h3>{category}</h3>
			{isLoading && <h2>Loading...</h2>}
			<div className="card-grid">
				{gifs?.map((image) => (
					// <li key={id}>
					// 	{/* <img src={gif.url} alt={gif.title} /> */}
					// 	{title}
					// </li>
					<GifItem key={image.id} {...image} />
				))}
			</div>
		</>
	);
};
