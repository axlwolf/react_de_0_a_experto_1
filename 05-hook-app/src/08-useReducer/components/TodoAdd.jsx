/* eslint-disable react/prop-types */
import { useForm } from "../../hooks/useForm";

export const TodoAdd = ({ handleNewTodo }) => {
	const { description, onInputChange, onResetForm } = useForm({
		description: "",
	});

	const onSubmittedTodo = (e) => {
		e.preventDefault();
		if (description.length <= 1) return;
		const newTodo = {
			id: new Date().getTime(),
			description,
			done: false,
		};
		handleNewTodo(newTodo);
		onResetForm();
	};

	return (
		<>
			<form onSubmit={onSubmittedTodo}>
				<input
					className="form-control"
					type="text"
					name="description"
					placeholder="Que hay que hacer?"
					onChange={onInputChange}
					value={description}
				/>

				<button type="submit" className="btn btn-outline-primary mt-2">
					Add
				</button>
			</form>
		</>
	);
};

//export default TodoAdd;
