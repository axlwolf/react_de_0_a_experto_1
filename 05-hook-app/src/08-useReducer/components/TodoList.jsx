import TodoItem from "./TodoItem";

/* eslint-disable react/prop-types */
const TodoList = ({ todos = [], onDeleteTodo, onToggleTodo }) => {
	return (
		<>
			<ul className="list-group">
				{todos.map((todo) => (
					<TodoItem
						key={todo.id}
						todo={todo}
						onDeleteTodo={onDeleteTodo}
						onToggleTodo={onToggleTodo}
					></TodoItem>
				))}
			</ul>
		</>
	);
};

export default TodoList;
