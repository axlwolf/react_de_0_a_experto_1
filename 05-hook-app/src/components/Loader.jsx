/* eslint-disable react/prop-types */
export const Loader = ({ isLoading }) => {
	if (isLoading)
		return <div className="alert alert-info text-center">Loading...</div>;
};
