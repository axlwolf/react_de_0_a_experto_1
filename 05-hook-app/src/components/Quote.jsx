import { useLayoutEffect, useRef, useState } from "react";

/* eslint-disable react/prop-types */
export const Quote = ({ sentence, character }) => {
	const pRef = useRef();

	const [boxSize, setBoxSize] = useState({ width: 0, height: 0 });

	useLayoutEffect(() => {
		console.log(pRef.current.getBoundingClientRect());
		setBoxSize({
			width: pRef.current.offsetWidth,
			height: pRef.current.offsetHeight,
		});
	}, [sentence]);

	return (
		<>
			<blockquote className="blockquote text-end" style={{ display: "flex" }}>
				<p ref={pRef} className="mb-1">
					{sentence}
				</p>
				<footer className="blockquote-footer">{character?.name}</footer>
			</blockquote>
			<code>{JSON.stringify(boxSize)}</code>
		</>
	);
};
