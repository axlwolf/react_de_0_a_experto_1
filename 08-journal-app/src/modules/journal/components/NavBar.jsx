/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import { useDispatch, useSelector } from "react-redux";

import { AppBar, IconButton, Grid, Toolbar, Typography } from "@mui/material";
import { MenuOutlined, LogoutOutlined } from "@mui/icons-material";
import { startLogOut } from "../../../store/auth";

export const NavBar = ({ drawerWidth = 240 }) => {
	const dispatch = useDispatch();

	const onLogOut = () => {
		console.log("Logout");
		dispatch(startLogOut());
	};

	return (
		<AppBar
			position="fixed"
			sx={{
				width: { sm: `calc(100% - ${drawerWidth}px)` },
				ml: { sm: `${drawerWidth}px` },
			}}
		>
			<Toolbar>
				<IconButton
					color="inherit"
					edge="start"
					sx={{ mr: 2, display: { sm: "none" } }}
				>
					<MenuOutlined></MenuOutlined>
				</IconButton>
				<Grid
					container
					direction="row"
					justifyContent="space-between"
					alignItems="center"
				>
					<Typography variant="h6" noWrap component="div">
						JournalApp
					</Typography>
					<IconButton color="error" onClick={onLogOut}>
						<LogoutOutlined></LogoutOutlined>
					</IconButton>
				</Grid>
			</Toolbar>
		</AppBar>
	);
};
