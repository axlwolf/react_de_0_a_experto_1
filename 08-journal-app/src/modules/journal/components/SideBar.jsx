/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import { TurnedInNot } from "@mui/icons-material";
import {
	Grid,
	Drawer,
	Box,
	Toolbar,
	Typography,
	Divider,
	List,
	ListItem,
	ListItemButton,
	ListItemIcon,
	ListItemText,
} from "@mui/material";
import { useSelector } from "react-redux";

export const SideBar = ({ drawerWidth = 240 }) => {
	const { displayName } = useSelector((state) => state.auth);

	return (
		<Box
			component="nav"
			sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
		>
			<Drawer
				variant="permanent"
				open
				sx={{
					display: { xs: "block" },
					"& .MuiDrawer-paper": { boxSizing: "border-box", width: drawerWidth },
				}}
			>
				<Toolbar>
					<Typography variant="h6" noWrap component="div">
						{displayName}
					</Typography>
				</Toolbar>
				<Divider />
				<List>
					{["Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio", "Julio"].map(
						(month) => (
							<ListItem key={month} disablePadding>
								<ListItemButton>
									<ListItemIcon>
										<TurnedInNot></TurnedInNot>
									</ListItemIcon>
									<Grid container>
										<ListItemText primary={month}></ListItemText>
										<ListItemText
											secondary={"Lorem ipsum, dolor sit amet consectetur."}
										></ListItemText>
									</Grid>
								</ListItemButton>
							</ListItem>
						)
					)}
				</List>
			</Drawer>
		</Box>
	);
};
