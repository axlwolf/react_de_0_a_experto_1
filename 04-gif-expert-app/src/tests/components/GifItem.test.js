/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe("Pruebas en <GifItem />", () => {
	const url = `https://media2.giphy.com/media/19JSJ5ucu91R5D7a3w/giphy.gif?cid=f1441f6f4s1k4db6p3d0oe0o5jks2ihdxv2gukm0hr3ma0j5&ep=v1_gifs_search&rid=giphy.gif&ct=g`;
	const title = "Dragon Ball Ultra Instinct GIF by Toei Animation";

	test("debe de hacer match con el snapshot", () => {
		const { container } = render(<GifItem title={title} url={url} />);

		expect(container).toMatchSnapshot();
	});
});
