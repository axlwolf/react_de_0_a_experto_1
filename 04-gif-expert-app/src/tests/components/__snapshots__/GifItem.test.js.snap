// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball Ultra Instinct GIF by Toei Animation"
      src="https://media2.giphy.com/media/19JSJ5ucu91R5D7a3w/giphy.gif?cid=f1441f6f4s1k4db6p3d0oe0o5jks2ihdxv2gukm0hr3ma0j5&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Dragon Ball Ultra Instinct GIF by Toei Animation
    </p>
  </div>
</div>
`;
